### ISSUE TITLE
`quote-props` rule false positive on computed properties.

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#3557]
Spawned from #3544.

This rule should ignore computed properties.

```js
/* eslint-env es6 */
/* eslint quote-props: [2, "always"] */
var x;
var obj = ({
    [x]: 0
});
```

Actual:

```
> eslint -v
v1.2.1
> eslint test.js --no-eslintrc

test.js
  5:5  error  Unquoted property `x` found  quote-props

✖ 1 problems (2 errors, 0 warnings)
```

Expected:

no errors.

**Comment by BYK** [comment]
Looking into this.

**Comment by mysticatea** [comment]
Thank you!
